### PR TITLE
EDM-296: Simplify helm charts and enable global helm values

### PR DESCRIPTION
**Pull request by rawagner**
Notable changes:
 - look at `global` values. This will be useful once the UI charts will be used as a sub-charts of the main FCTL chart
 - namespace is taken from `.Release.Namespace`
 - non-secret values are stored in `ConfigMap` instead of `Secret`
 - qcow2 and bootc vars are removed as the feature is simply not there/worth exposing

This is a first part of being able to include UI chart as sub-chart to main FCTL chart

### CODE REVIEW

File: apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml
Comment (celdrake) [inline apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml]: I tested it locally and the terminal worked fine, but we were previously using port 7444.
Would port 7443 work also in the deployed environment?

File: apps/standalone/deploy/helm/flightctl-ui/Chart.yaml
Comment (celdrake) [inline apps/standalone/deploy/helm/flightctl-ui/Chart.yaml:5]: Should we maybe up the `version`? (Same for the rest of the OCP plugin chart)

File: apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml
Comment (celdrake) [inline apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml:7]: We'd need to make changes to the "Add devices" modal if we're not supporting any option for adding devices, and checkout the UX with Nir.

And ideally also remove the references to the two properties in the rest of the codebase (proxy, AppContext, etc)

File: apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/templates/flightctl-ui-deployment.yaml
Comment (mangelajo) [inline apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/templates/flightctl-ui-deployment.yaml]: 👍

File: apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml
Comment (mangelajo) [reply apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml]: good catch, it's 7444

File: apps/standalone/deploy/helm/flightctl-ui/Chart.yaml
Comment (mangelajo) [reply apps/standalone/deploy/helm/flightctl-ui/Chart.yaml:5]: 1. When we do the helm publish via an OCI container, it's injected into the Chart.yaml

File: apps/standalone/deploy/helm/flightctl-ui/templates/flightctl-ui-certs-secret.yaml
Comment (mangelajo) [inline apps/standalone/deploy/helm/flightctl-ui/templates/flightctl-ui-certs-secret.yaml]:
```suggestion
  ca_oidc.crt: {{ default (.Values.global).flightctl.auth.caCert .Values.auth.caCert | quote }}
```
Just curious, why the braces? are those mandatory?

File: apps/standalone/deploy/helm/flightctl-ui/values.yaml
Comment (mangelajo) [inline apps/standalone/deploy/helm/flightctl-ui/values.yaml]:
```suggestion
auth:
  internalOidcAuthority: ""
  insecureSkipTlsVerify: false
  caCert: ""
```

We should converge to put the auth details as globals, because I see the API also uses those.

https://github.com/flightctl/flightctl/blob/main/deploy/helm/flightctl/values.yaml#L65

What about having all of them as global.flightctl.auth   ?

File: apps/standalone/deploy/helm/flightctl-ui/values.yaml
Comment (mangelajo) [inline apps/standalone/deploy/helm/flightctl-ui/values.yaml]: We should probably document any global parameters being consumed.

File: apps/standalone/deploy/helm/flightctl-ui/values.yaml
Comment (mangelajo) [inline apps/standalone/deploy/helm/flightctl-ui/values.yaml]:
```suggestion
  grpcUrl: grpcs://flightctl-api-agent-grpc:7444/
```

File: apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml
Comment (mangelajo) [inline apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml]: let's document also any global consumed values.

File: apps/standalone/deploy/helm/flightctl-ui/values.yaml
Comment (rawagner) [reply apps/standalone/deploy/helm/flightctl-ui/values.yaml]: Yes, I expect that we will provide these through globals. See that in the template I am using 
```
ca_oidc.crt: {{ default (.Values.global).flightctl.auth.caCert .Values.auth.caCert | quote }}
```

but I also add these in separate UI auth object in order to allow specifying different auth config than the global one. 
As maybe globally we want to use some `internalOidcAuthority` but for UI we want a different one. Im not sure why, but I think we want this flexibility.  WDYT ?

File: apps/standalone/deploy/helm/flightctl-ui/templates/flightctl-ui-certs-secret.yaml
Comment (rawagner) [reply apps/standalone/deploy/helm/flightctl-ui/templates/flightctl-ui-certs-secret.yaml]: if `global` object is not defined we wont fail here. Without braces, the helm template fails with nil pointer exception

File: apps/standalone/deploy/helm/flightctl-ui/values.yaml
Comment (mangelajo) [reply apps/standalone/deploy/helm/flightctl-ui/values.yaml]: That would work, we just need to document the globals then :)

The bad side of adding too many knobs like this (global vs local) is that we need to maintain them later, I was considering on removing a lot of the knobs on the backend may be, but had not decided. 

What do you think? is there a possibility that we may need different oidc from the UI compared to the backend?

Comment (rawagner) [reply apps/standalone/deploy/helm/flightctl-ui/values.yaml]: Yeah, the bad side here is that we need to maintain these ..
Lets imagine a case where you deploy
 - FCLT API and Keycloak to namespace FOO
 - FCTL UI to namespace BAR
 
 then you could set `global.auth.internalOIDC` to `https://keycloak` and UI auth will need to be `https://keycloak.FOO`

or you could set global to `https://keycloak.FOO` and it will work for both  ...
(but for this we would need to enable deploying UI to a different NS than API)

I guess it mostly depends on what we agree :)

Comment (mangelajo) [reply apps/standalone/deploy/helm/flightctl-ui/values.yaml]: Yeah, good thinking,

It would make sense to agree to document that when a global.internalOidcAuthority is set, we must provide the full cluster path for the URL, including the namespace, and the same for any other internal endpoints. Then, it may work in all cases. We should probably do it regardless of keeping the specific knobs.


Another question I have is if we could auto-calculate that knob value if keycloak is enabled in the deployment. May be that is just some further refinement we could think of.

File: apps/standalone/deploy/helm/flightctl-ui/templates/flightctl-ui-certs-secret.yaml
Comment (mangelajo) [reply apps/standalone/deploy/helm/flightctl-ui/templates/flightctl-ui-certs-secret.yaml]: nice I didn't know about that.

File: apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml
Comment (mangelajo) [reply apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml]: This one is still pending in the same way we have for the other values:

https://github.com/flightctl/flightctl-ui/pull/112/files#diff-9712335e7da43c32d9689c43790084c6645fdaa287062d9f2b4eac266dbc07e1R6

Comment (mangelajo) [reply apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml]: Or we don't need any of those at all?

Comment (rawagner) [reply apps/ocp-plugin/deploy/helm/flightctl-ocp-ui/values.yaml]: Not at the moment. Maybe later we will add some, but currently I dont see any that would be useful here.

**Review by mangelajo — APPROVED** [review]

**Comment by celdrake** [comment]
@rawagner The fix for the Cypress tests is https://github.com/flightctl/flightctl-ui/pull/115/commits/59eefdb3514af53a9b1f424755472cd8f9928c88. 

I can extract it to a separate PR if you prefer, so we can merge this PR.

**Comment by mangelajo** [comment]
> @rawagner The fix for the Cypress tests is [59eefdb](https://github.com/flightctl/flightctl-ui/commit/59eefdb3514af53a9b1f424755472cd8f9928c88).
> 
> I can extract it to a separate PR if you prefer, so we can merge this PR.

The tests aren't mandatory yet. So you could technically merge them individually, in the end you know what you are doing and this PR didn't break the integration tests I believe.

**Comment by celdrake** [comment]
> > @rawagner The fix for the Cypress tests is [59eefdb](https://github.com/flightctl/flightctl-ui/commit/59eefdb3514af53a9b1f424755472cd8f9928c88).
> > I can extract it to a separate PR if you prefer, so we can merge this PR.
> 
> The tests aren't mandatory yet. So you could technically merge them individually, in the end you know what you are doing and this PR didn't break the integration tests I believe.

Well, the merge button does not become enabled unless the CI step passes. The tests were not broken in this PR, so I'd rather fix them so we can get green builds and merge necessary PRs. I'm doing this now here https://github.com/flightctl/flightctl-ui/pull/117

**Comment by rawagner** [comment]
Thanks @celdrake for fixing the tests. I will mark this as a draft as I'd like to get an agreement on the backend charts before merging this. I will be opening the PR for backend shortly

**Comment by rawagner** [comment]
Backend PR https://github.com/flightctl/flightctl/pull/554

**Comment by mangelajo** [comment]
@rawagner is this still draft? I am reviewing.

File: deploy/helm/flightctl-ui/templates/flightctl-ui-deployment.yaml
Comment (mangelajo) [inline deploy/helm/flightctl-ui/templates/flightctl-ui-deployment.yaml:67]: 🎸

File: deploy/helm/flightctl-ui/templates/flightctl-ui-service.yaml
Comment (mangelajo) [inline deploy/helm/flightctl-ui/templates/flightctl-ui-service.yaml:8]:
```suggestion
  {{- if not ((.Values.global).baseDomainTls).cert }}
```

Why is this inverted everywhere, shouldn't it be:

  {{- if ((.Values.global).baseDomainTls).cert }}

?

May be I just need more coffee ':D

Comment (mangelajo) [reply deploy/helm/flightctl-ui/templates/flightctl-ui-service.yaml:8]: also confused with the instances in this file : https://github.com/flightctl/flightctl-ui/pull/112/files#diff-325541283d9f28d1334be3088b93999822f1624d9b7a435c54c76c6aa4a7a564R79

Comment (rawagner) [reply deploy/helm/flightctl-ui/templates/flightctl-ui-service.yaml:8]: This condition means:
if user DOES NOT specify their own TLS cert (via `global.baseDomainTls`), we will annotate the Service and let OpenShift create a cert for us.

Comment (rawagner) [reply deploy/helm/flightctl-ui/templates/flightctl-ui-service.yaml:8]: You're right about the second one, the condition should not be there at all.

**Review by mangelajo — APPROVED** [review]
Looks good to me :)
time to try! :D

**Pull request merged**
